pkg/kube/utils.go

```diff
@@ -171,3 +174,20 @@ func PVCContainsReadOnlyAccessMode(pvc *corev1.PersistentVolumeClaim) bool {
 
 	return false
 }
+
+// AddDebugLabelsToPodOptions adds additional label selector to `PodOptions`,
+// provided the context has a key starting with `keyPrefix`.
+func AddDebugLabelsToPodOptions(ctx context.Context, options *PodOptions, keyPrefix, keySuffix string) {
```

Comment:
1. Since the values are extracted from context, the function name should suggest that. The function name should be self-explanatory.
2. Instead of passing prefix and suffix as a parameter to the function, its better to just pass key string. The key construction should be done by caller.
```suggestion
func AddLabelsToPodOptionFromContext(ctx context.Context, options *PodOptions, labelKey string) {
```

Reply:
Right, we shouldn't add this check in this utility function. The function should just do one thing - add labels in the option with the given key. The callers should do the validation on keys if required (maybe separate function for validateLabelKeys()).

Reply:
Done